Formula/flat.rb

```diff
@@ -0,0 +1,15 @@
+class Flat < Formula
+  desc "Flattens json or yaml structure"
+  homepage "https://github.com/tlopo-ruby/flat"
+  url "https://github.com/tlopo-ruby/flat.git", :using => :git, :revision => "bde2c6e47d055f7c7f27c06d6438bc2a6a6e3967"
+  version "0.0.2"
+
+  def install
+    mv "flat.rb", "flat"
+    bin.install "flat"
+  end
+
+  test do
+    assert_match "Usage: flat", shell_output("#{bin}/flat --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.